server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -566,6 +568,13 @@ final SearchContext createAndPutContext(ShardSearchRequest request) throws IOExc
 
     final SearchContext createContext(ShardSearchRequest request) throws IOException {
         final DefaultSearchContext context = createSearchContext(request, defaultSearchTimeout);
+        if (activeContexts.size() >= Node.MAX_SEARCH_CONTEXT_SETTING.get(settings)) {
```

Comment:
should we rather add this check at the beginning of `createAndPutContext`? This method only creates a new context, it doesn't add it to the list of active contexts?